exercise-book/src/tcp-server.md

```diff
@@ -0,0 +1,81 @@
+# Interactive TCP Echo Server
+
+In this exercise, we will make a simple TCP "echo" server using APIs in Rust standard Library. Here's how an interaction with it would look like from a client point of view. You connect to it using `nc`, for example:
+
+`nc localhost 7878`
+
+and type in one line of text. As soon as you hit enter, the server sends the line back but keeps the connection opened. You can type another line and get it back, and so on.
+
+Here's an example interaction with the server. Notice that after typing a single line the connection is not closed and we receive the line back. All inputs and outputs should be separated by new line characters (`\n`).
+
+```text
+$ nc localhost 7878
+hello
+> hello
+world
+> world
+```
+
+(`>` denotes the text that is send back to you)
+
+## After completing this exercise you are able to
+
+- open a TCP port and react to each user connects
+
+- use I/O traits to read/write from a TCP socket
+
+- use threads to support multiple connections
+
+## Tasks
+
+1. Create a new binary project `tcp-server`
+2. Implement a basic TCP server that listens for connections on a given port (you can use `127.0.0.1:7878` or any other port that you like).
+3. Implement a loop that would read data from a `TcpStream` one line at a time. We assume that lines are separated by a `\n` character.
+4. Add writing the received line back to the stream. Resolve potential borrow checker issues using standard library APIs.
+5. Use Rust's `thread` API to add support for multiple connections.
+
+Here's a bit of code to get you started:
+
+```rust
```

Comment:
I think you need to mark this `no_run` (or something like that), because it panics when you run it.